### PR TITLE
docs(cli): Improve help text

### PR DESCRIPTION
**Pull request by aparcar**
This commit does multiple things at once to improve the user experience:
* Always start with an upper case letter, no mixing
* Hide deprecated `keygen` command, it still works if called
* Extend and rework some documentation textx
* Drop false `log_level` text, it contains a logic error
* Wrap all documentation at 80 chars

### CODE REVIEW

File: rosenpass/src/cli.rs [hunk 1]
```diff
@@ -194,19 +205,19 @@ pub enum CliCommand {
         force: bool,
     },
 
-    /// Generate the keys mentioned in a configFile
+    /// Generate secret- & public-key for Rosenpass
```
Comment (blipp) [inline rosenpass/src/cli.rs:208]:
```suggestion
    /// Generate secret & public key for Rosenpass
```

File: rosenpass/src/cli.rs [hunk 2]
```diff
@@ -194,19 +205,19 @@ pub enum CliCommand {
...
     ///
-    /// Generates secret- & public-key to their destination. If a config file
-    /// is provided then the key file destination is taken from there.
-    /// Otherwise the
+    /// Generates secret- & public-key to their destination. If a config file is
```
Comment (blipp) [inline rosenpass/src/cli.rs:210]:
```suggestion
    /// Generates secret & public key to their destination. If a config file is
```

File: rosenpass/src/cli.rs [hunk 3]
```diff
@@ -194,19 +205,19 @@ pub enum CliCommand {
...
+    /// provided then the key file destination is taken from there, otherwise
+    /// the destination is taken from the CLI arguments.
     GenKeys {
         config_file: Option<PathBuf>,
 
-        /// where to write public-key to
+        /// Where to write public-key to
```
Comment (blipp) [inline rosenpass/src/cli.rs:216]:
```suggestion
        /// Where to write public key to
```

File: rosenpass/src/cli.rs [hunk 4]
```diff
@@ -194,19 +205,19 @@ pub enum CliCommand {
...
         #[clap(short, long)]
         public_key: Option<PathBuf>,
 
-        /// where to write secret-key to
+        /// Where to write secret-key to
```
Comment (blipp) [inline rosenpass/src/cli.rs:220]:
```suggestion
        /// Where to write secret key to
```

File: rosenpass/src/cli.rs [hunk 5]
```diff
@@ -143,28 +149,29 @@ impl CliArgs {
...
     /* Explanation: `first_arg` and `rest_of_args` are combined into one
      * `Vec<String>`. They are only used to trick clap into displaying some
      * guidance on the CLI usage.
      */
     #[allow(rustdoc::broken_intra_doc_links)]
     #[allow(rustdoc::invalid_html_tags)]
     Exchange {
-        /// public-key <PATH> secret-key <PATH> [listen <ADDR>:<PORT>]... [verbose]
+        /// public-key <PATH> secret-key <PATH> [listen <ADDR>:<PORT>]...
+        /// [verbose]
```
Comment (blipp) [inline rosenpass/src/cli.rs:174]: This change wraps command-line arguments. Does this end up in an alright format when rendered in various formats? Independent of the wrapping, does this need backticks?

File: rosenpass/src/cli.rs [hunk 6]
```diff
@@ -176,7 +183,8 @@ pub enum CliCommand {
         ///
         /// OUTFILE := outfile <PATH>
         ///
-        /// WG := wireguard <WIREGUARD_DEV> <WIREGUARD_PEER> [WIREGUARD_EXTRA_ARGS]...
+        /// WG := wireguard <WIREGUARD_DEV> <WIREGUARD_PEER>
+        /// [WIREGUARD_EXTRA_ARGS]...
```
Comment (blipp) [inline rosenpass/src/cli.rs:187]: Some comment than above: This change wraps command-line arguments. Does this end up in an alright format when rendered in various formats? Independent of the wrapping, does this need backticks?

**Review by blipp — CHANGES_REQUESTED** [review]

**Comment by blipp** [comment]
> * Hide deprecated `keygen` command, it still works if called

I didn't see where this is done in this set of changes, could you please help me out?

File: rosenpass/src/cli.rs (hunk 5 above)
Comment (aparcar) [reply rosenpass/src/cli.rs:174]: It's rendered correctly and does not need backticks. However I agree that his is not super pretty, long term I'm again this kind of parameter passing but that's another discussion

File: rosenpass/src/cli.rs (hunk 6 above)
Comment (aparcar) [reply rosenpass/src/cli.rs:187]: Same answer as above https://github.com/rosenpass/rosenpass/pull/486#discussion_r1838207339 :)

File: rosenpass/src/cli.rs [hunk 7]
```diff
@@ -215,21 +224,27 @@ pub enum CliCommand {
...
+    Validate { config_files: Vec<PathBuf> },
+
+    /// DEPRECATED - use the gen-keys command instead
     #[allow(rustdoc::broken_intra_doc_links)]
     #[allow(rustdoc::invalid_html_tags)]
+    #[command(hide = true)]
     Keygen {
-        // NOTE yes, the legacy keygen argument initially really accepted "privet-key", not "secret-key"!
+        // NOTE yes, the legacy keygen argument initially really accepted
+        // "privet-key", not "secret-key"!
```
Comment (blipp) [inline rosenpass/src/cli.rs:240]:
```suggestion
        // NOTE yes, the legacy keygen argument initially really accepted
        // "private-key", not "secret-key"!
```

I assume this is just a long-standing typo in the comment, and that the CLI actually needed `private-key`? 😅

**Review by blipp — CHANGES_REQUESTED** [review]

File: rosenpass/src/cli.rs (hunk 7 above)
Comment (aparcar) [reply rosenpass/src/cli.rs]: Fixed.

**Pull request merged**